### PR TITLE
RavenDB-14309 - Cannot start the server if the cluster is down

### PR DESCRIPTION
**Pull request by grisha-kotler**
https://issues.hibernatingrhinos.com/issue/RavenDB-14309

### CODE REVIEW

**Review by ayende — APPROVED** [review]

File: src/Raven.Server/Monitoring/Snmp/SnmpWatcher.cs [hunk 1]
```diff
@@ -382,8 +392,16 @@ private async Task AddDatabases()
...
-                        var result = await _server.ServerStore.SendToLeaderAsync(new UpdateSnmpDatabasesMappingCommand(missingDatabases, RaftIdGenerator.NewId()));
-                        await _server.ServerStore.Cluster.WaitForIndexNotification(result.Index);
+                        try
+                        {
+                            var result = await _server.ServerStore.SendToLeaderAsync(new UpdateSnmpDatabasesMappingCommand(missingDatabases, RaftIdGenerator.NewId()));
+                            await _server.ServerStore.Cluster.WaitForIndexNotification(result.Index);
+                        }
+                        catch (Exception e) when (e is TimeoutException || e is OperationCanceledException)
+                        {
+                            // we will do the update in AddDatabaseIfNecessary
```
Comment (arekpalinski) [inline src/Raven.Server/Monitoring/Snmp/SnmpWatcher.cs]: Which is hooked to OnDatabaseLoaded event, so we'll do it on the actual database load, right?

File: src/Raven.Server/Monitoring/Snmp/SnmpWatcher.cs [hunk 2]
```diff
@@ -394,6 +412,11 @@ private async Task AddDatabases()
                         LoadDatabase(database, mapping[database]);
                 }
             }
+            catch (Exception e)
+            {
+                if (Logger.IsOperationsEnabled)
+                    Logger.Operations("Failed to update the SNMP mapping", e);
```
Comment (arekpalinski) [inline src/Raven.Server/Monitoring/Snmp/SnmpWatcher.cs]: Can we add database names we tried? - `missingDatabases`

File: src/Raven.Server/Monitoring/Snmp/SnmpWatcher.cs (hunk 1 above)
Comment (grisha-kotler) [reply src/Raven.Server/Monitoring/Snmp/SnmpWatcher.cs]: Yes. Updated the comment

File: src/Raven.Server/Monitoring/Snmp/SnmpWatcher.cs (hunk 2 above)
Comment (grisha-kotler) [reply src/Raven.Server/Monitoring/Snmp/SnmpWatcher.cs]: done

**Pull request merged**